perf/impl/js-libp2p/v0.46/Makefile

```diff
@@ -0,0 +1,25 @@
+NPM_PACKAGE_VERSION := 1.1.1
+NPM_PACKAGE := @libp2p/perf
+NPM_PACKAGE_NAME := $(NPM_PACKAGE)@$(NPM_PACKAGE_VERSION)
+
+SOURCE_DIR := js-libp2p-protocol-perf
+
+DOCKER_IMAGE := node:18.17.1
+DOCKER_RUN := docker run --rm -v "$(shell pwd)/$(SOURCE_DIR)":/usr/src/myapp -w /usr/src/myapp $(DOCKER_IMAGE)
+
+all: perf
+
+perf: install_node
+	mkdir -p $(SOURCE_DIR)
+	$(DOCKER_RUN) npm install $(NPM_PACKAGE_NAME)
+
+install_node:
+	sudo yum update -y
+	curl -fsSL https://rpm.nodesource.com/setup_lts.x | sudo bash -
+	sudo yum install -y nodejs
+	node -v
+	npm -v
```

Comment:
The `install_node` target alters state beyond this repository folder, i.e. it alters global state, the systems NodeJS installation. In addition it assumes that it is running on a RPM based Linux distribution. Neither is ideal. As a user of the libp2p perf tooling when I run `cd perf/ && make` I expect all changes to be limited to this folder (ignoring caches). As a user of the libp2p perf tooling, I expect to be able to run `cd perf/ && make` ideally on a Mac, though at least on any Linux distribution, not just the RPM ones.

Can we do better? Ideas that come to my mind:

- We could run the server and client in Docker. Though that might have a performance impact which is the reason we run all the other implementations outside of Docker.
- Install NodeJS globally on the machine (`user-data.sh`). Downside is that different js-libp2p versions can not use different NodeJS versions.
- Use tooling like [pkg](https://github.com/vercel/pkg).

I have little to no experience with packaging NodeJS applications. @maschad @achingbrain @MarcoPolo do you have better ideas?

Reply:
Well the assumption is that this `Makefile` is intended to be run on the Terraform infrastructure which at the moment is AWS, in this specific case, EC2 instances. This is also the same premise for the `perf` bash script which calls [upon node to start the server/client](https://github.com/libp2p/test-plans/pull/244/files#diff-d634d23030fcd1a8ce7d7daa1cc2431802c5ff307bd84652786e9059925022c7R33).

These `Makefile`s across the implementations aren't really useful outside of the use case of [running the benchmark results](https://github.com/libp2p/test-plans/blob/perf/js-libp2p/perf/runner/src/index.ts#L219), which would rely on the aforementioned Terraform infrastructure, and so I don't see a rationale for a user running `cd perf/impl/ && make` locally.

Thus tailoring the `Makefile` to an RPM Linux distro and altering global state should be fine imo considering these instances will be destroyed once the infrastructure is tore down after a test run.